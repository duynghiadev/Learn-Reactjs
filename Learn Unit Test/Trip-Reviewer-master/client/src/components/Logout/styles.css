.logout-container h1 {
  color: #02bcd4;
  font-size: 30px;
  font-weight: 300;
  text-align: center;
  padding-top: 31px;
}
