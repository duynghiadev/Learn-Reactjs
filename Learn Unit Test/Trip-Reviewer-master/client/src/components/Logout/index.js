import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';

import './styles.css';

class Logout extends Component {
  componentWillMount() {
    axios.get('/api/logout').then(this.handleResponce);
  }

  handleResponce = () => {
    setTimeout(this.redirectToHome, 2000);
  };

  redirectToHome = () => {
    this.props.history.push('/');
  };

  render() {
    return (
      <div className="logout-container">
        <h1>Sorry to see you go :(</h1>
      </div>
    );
  }
}

Logout.propTypes = {
  history: PropTypes.object.isRequired,
};

export default Logout;
