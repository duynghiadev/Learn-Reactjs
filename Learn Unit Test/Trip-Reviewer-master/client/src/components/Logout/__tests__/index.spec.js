import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';

import Logout from '..';

const mockComponent = props => <Logout {...props} />;

describe('<Logout />', () => {
  jest.useFakeTimers();

  it('should render component', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should redirect to home after some time', () => {
    const props = {
      history: {
        push: jest.fn(),
      },
    };
    const instance = mount(mockComponent(props)).instance();

    instance.redirectToHome();

    expect(props.history.push).toHaveBeenCalledWith('/');
  });

  it('should handle request correctly', () => {
    const instance = mount(mockComponent({})).instance();

    instance.handleResponce();

    expect(setTimeout).toHaveBeenCalledWith(instance.redirectToHome, 2000);
  });
});
