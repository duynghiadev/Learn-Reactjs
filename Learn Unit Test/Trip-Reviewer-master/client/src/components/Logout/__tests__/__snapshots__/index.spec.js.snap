// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> should render component 1`] = `
<div
  className="logout-container"
>
  <h1>
    Sorry to see you go :(
  </h1>
</div>
`;
